### PR TITLE
Added 2 missing researches

### PR DESCRIPTION
**Pull request by Sfiguz7**
## Description

Researches for tier II electric machines and coal/lava gens were missing. I added them in
## Changes

Added 2 researches, moved Electric Ore Grinder II to the tier II machines research
## Related Issues

N/A
## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/setup/ResearchSetup.java [hunk 1]
```diff
@@ -283,6 +284,7 @@ public static void setupResearches() {
         register("wise_talisman", 271, "Talisman of the Wise", 20, SlimefunItems.TALISMAN_WISE);
         register("book_binder", 272, "Enchantment Book Binding", 26, SlimefunItems.BOOK_BINDER);
         register("auto_crafting", 273, "Automatic Crafting", 30, SlimefunItems.VANILLA_AUTO_CRAFTER, SlimefunItems.ENHANCED_AUTO_CRAFTER);
+        register("improved_generators", 274, "Improved Generators", 38, SlimefunItems.COAL_GENERATOR_2, SlimefunItems.LAVA_GENERATOR_2);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/setup/ResearchSetup.java:287]: I think 38 is hella overpriced as a default value.

```suggestion
        register("improved_generators", 274, "Improved Generators", 24, SlimefunItems.COAL_GENERATOR_2, SlimefunItems.LAVA_GENERATOR_2);
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Comment by TheBusyBiscuit** [comment]
Also, you forgot to add the names to the `researches_en.yml` :LUL:
Thank my past self for making a Unit Test to spot this :P

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/setup/ResearchSetup.java (hunk 1 above)
Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/setup/ResearchSetup.java:287]: Lava gen research is 38, I made it same cost. Should I lower the tier 1 lava gen then?

Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/setup/ResearchSetup.java:287]: I'll take it as a yes

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/setup/ResearchSetup.java [hunk 2]
```diff
@@ -283,6 +284,8 @@ public static void setupResearches() {
         register("wise_talisman", 271, "Talisman of the Wise", 20, SlimefunItems.TALISMAN_WISE);
         register("book_binder", 272, "Enchantment Book Binding", 26, SlimefunItems.BOOK_BINDER);
         register("auto_crafting", 273, "Automatic Crafting", 30, SlimefunItems.VANILLA_AUTO_CRAFTER, SlimefunItems.ENHANCED_AUTO_CRAFTER);
+        register("improved_generators", 274, "Improved Generators", 24, SlimefunItems.COAL_GENERATOR_2, SlimefunItems.LAVA_GENERATOR_2);
+        register("ingredients_and_cheese", 275, "Slimefun Cuisine", 5, SlimefunItems.SALT, SlimefunItems.WHEAT_FLOUR, SlimefunItems.HEAVY_CREAM, SlimefunItems.CHEESE, SlimefunItems.BUTTER);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/setup/ResearchSetup.java:288]: I think these don't really need a research but okay...

**Review by TheBusyBiscuit — APPROVED** [review]
LGTM

File: src/main/resources/languages/researches_en.yml [hunk 3]
```diff
@@ -250,4 +251,9 @@ slimefun:
   bee_armor: Bee Armor
   book_binder: Enchantment Book Binding
   auto_crafting: Automatic Crafting
+<<<<<<< HEAD
+  improved_generators: Improved Generators
+  ingredients_and_cheese: Slimefun Cuisine
+=======
   produce_collector: Automatic Milking
+>>>>>>> 449a2524e91a46bc5d5e82980ad0f5f18f2e0ef4
```
Comment (TheBusyBiscuit) [inline src/main/resources/languages/researches_en.yml:259]: Good job resolving these conflicts :NotLikeThis:

Comment (Sfiguz7) [reply src/main/resources/languages/researches_en.yml:259]: Yea I noticed :(

File: src/main/resources/languages/researches_en.yml [hunk 4]
```diff
@@ -251,3 +252,6 @@ slimefun:
   book_binder: Enchantment Book Binding
   auto_crafting: Automatic Crafting
   produce_collector: Automatic Milking
+  improved_generators: Improved Generators
+  ingredients_and_cheese: Slimefun Cuisine
+
```
Comment (WalshyDev) [inline src/main/resources/languages/researches_en.yml:257]:
```suggestion
```

Comment (Sfiguz7) [reply src/main/resources/languages/researches_en.yml]: Really now. Really.

**Review by WalshyDev — APPROVED** [review]
LGTM

**Review by TheBusyBiscuit — APPROVED** [review]

**Pull request merged**